zh/basic_concepts/scene.md

```diff
@@ -1,75 +1,46 @@
-</div>
+# 场景(Scene)
 
-## Scene
-In your game you probably want a main menu, a few levels and an ending scene.
-How do you organize all of these into the separate pieces they are? You guessed
-it, `Scene`. When you think about your favorite movie you can see that it's
-distinctly broken down into scenes, or separate parts of the story line. If we
-apply this same thought process to games, we should come up with at least a few
-scenes no matter how simple the game is.
+在游戏开发过程中，你可能需要一个主菜单，几个关卡和一个结束场景。如何组织所有这些分开的部分？使用 __场景(Scene)__ ! 当你想到喜欢的电影时，你能观察到它是被分解为不同场景或不同故事线。现在我们对游戏开发应用这个相同的思维过程，你应该很容易就能想出几个场景。
 
-Taking another look at the familiar image from earlier:
+来看一张熟悉的图片：
 
-![](basic_concepts-img/2n_main.png "")
+![](../../en/basic_concepts/basic_concepts-img/2n_main.png "")
 
-This is a main menu and it is a single `Scene`. This scene is made up of
-several pieces that all fit together to give us the end result. Scenes are drawn
-by the __renderer__. The __renderer__ is responsible for rendering sprites and
-other objects into the screen. To better understand this we need to talk a bit
-about the __scene graph__.
+这是一个主菜单场景, 这个场景是由很多小的对象拼接而成, 所有的对象组合在一起, 形成了最终的结果. 场景是被 __渲染器(renderer)__ 画出来的. 渲染器负责渲染精灵和其它的对象进入屏幕. 为了更好的理解这个过程, 我们需要讨论一下 __场景图__.
 
-### Scene Graph
-A __scene graph__ is a data structure that arranges a graphical scene. A
-__scene graph__ contains `Node` objects in a tree (yes, it is called
-__scene graph__, but it is actually represented by a __tree__) structure.
+## 场景图(Scene Graph)
+
+场景图(Scene Graph)是一种安排场景内对象的数据结构, 它把场景内所有的 __节点(Node)__ 都包含在一个 __树(tree)__ 上. (场景图虽然叫做"图",但实际使用一个树结构来表示).
 
 ![](basic_concepts-img/tree.jpg "Simple Tree")
 
-It sounds and looks complicated. I'm sure you are asking why should you care
-about this technical detail if Cocos2d-x does the heavy lifting for you? It
-really is important to understand how Scenes are drawn by the renderer.
+听起来这好像很复杂, 可能你会问, 我为什么要关注这个技术细节, Cocos2d-x 值得我研究的这么深入吗? 值得! 这个对你真正了解渲染器是如何绘制场景的非常重要. 
 
-Once you start adding nodes, sprites and animations to your game, you want to
-make sure you are drawing the things you expect. But what if you are not?  What
-if your sprites are hidden in the background and you want them to be the
-foremost objects? No big deal, just take a step back and run through the scene
-graph on a piece of paper, and I bet you find your mistake easily.
+当你开发游戏的时候, 你会添加一些节点, 精灵和动画到一个场景中, 你期望的是每一个添加的对象都能被正确的展示, 可是如果有个对象没有被展示呢? 可能你错误的把这个对象隐藏到背景中了. 怎么办? 别着急, 这是个小问题, 停下来, 拿出一张纸, 把场景图画出来, 你肯定能很容易的发现错误.
 
 Since the _Scene Graph_ is a tree; you can __walk the tree__. Cocos2d-x uses
 the __in-order walk__ algorithm. An __in-order walk__ is the left side of the
 tree being walked, then the root node, then the right side of the tree. Since
 the right side of the tree is rendered last, it is displayed first on the
-__scene graph__.
+__scene graph__. `错误描述 从 A-F 的顺序来看, 是按照左右根的顺序, 不是左根右`
 
-![](basic_concepts-img/in-order-walk.png "in-order walk")
```

Comment:
there is some doubt about this paragraph, it's not consistent with `basic_concepts-img/in-order-walk.png`, the image shows order A,B...F order by `tree left, then tree right, then tree root`

Reply:
I have seen the wiki site from api: https://en.wikipedia.org/wiki/Tree_traversal#In-order, this image is not clearly to show in-order, I think you should add that you will get order `A, B, C, D, E, F, G, H, I` following in-order. for the same image you will get `F, B, A, D, C, E, G, I, H.` following pre-order

Reply:
Lets ask @minggo because this diagram was approved by Ricardo. http://cocos2d-x.org/docs/programmers-guide/basic_concepts-img/in-order-walk.png

Reply:
the graph is correct, engine will go through the children by

- children on left(zOrder < 0)
- self
- children on right(zOrder >= 0)

Reply:
you can find this diagram from the wiki: https://en.wikipedia.org/wiki/Tree_traversal#In-order, in-order, pre-order, post-order all have the same diagram. but in this docs we just want to say in-order, so I think we should add the result order, not only show this diagram. another question is Scene Graph is a kind of Binary tree or not

Reply:
@drelaptop misunderstood the graph, they are different.

Reply:
thank you, I was wrong